sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -200,14 +201,15 @@ object DataSourceV2Strategy extends Strategy with PredicateHelper {
             catalog,
             ident,
             parts,
+            query,
             planLater(query),
             props,
             writeOptions,
             orCreate = orCreate) :: Nil
       }
 
     case AppendData(r: DataSourceV2Relation, query, _) =>
-      AppendDataExec(r.table.asWritable, r.options, planLater(query)) :: Nil
+      AppendDataExec(r.table.asWritable, r.options, query, planLater(query)) :: Nil
```

Comment:
I think that falling back to v1 should use different plan nodes so that users can see that the v1 write API is used instead of v2. It would also help keep concerns separated in the exec nodes.

Reply:
That's gonna require a separation at the catalog level

Reply:
We talked about this separately and I think this is okay for now. In the future, we can add a separate `Write` produced by the builder (like `Scan` in the read side) and use that to use a separate plan.